### PR TITLE
Add tests for git status section

### PR DESCRIPTION
**Pull request by matchai**
## Description

The git status section has a fair bit of complexity and should probably be tested.
In the process of writing these tests, I found a few issues in other existing tests.

## Motivation and Context

Improve that test coverage! ⬆️ 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

- [x] I have tested using **MacOS**
- [ ] I have tested using **Linux**

## Checklist:

- [x] I have checked that no other PR duplicates mine
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

**Comment by Snuggle** [comment]
Are you stuck on anything in particular here? :blush:

**Comment by matchai** [comment]
Thanks for asking. 😄
Mostly the tests that I've mentioned in these comments:
https://github.com/matchai/spacefish/blob/792ba48382d2ae7ffc5311ebbd9256f51392185e/tests/__sf_section_git_status.test.fish#L120-L123

I was thinking of doing some work on our mock util to allow for more flexible mocks so that we can properly test these (e.g. specifying the command *and* argument necessary to trigger a mock).

**Comment by Snuggle** [comment]
Oh, heh. It might be better to have a checklist in the PR for to-do lists in the future! 😊

And I'd love such an improvement, would make things much easier!

**Comment by matchai** [comment]
Gah! After all this work rewriting our mocking library, I just realized that we use `command git` in the git section, so we can't mock it! 😖

@Snuggle Any ideas as to how we can achieve these test states? I'm considering just merging this in for now. It's definitely a step above what we used to have for git status testing.

**Comment by matchai** [comment]
I'll just go ahead and merge this in how it is, once I get a review. 😄

**Comment by Snuggle** [comment]
Could I ask what this `\*` syntax does? Is it globbing a bunch of files in a directory or...?

https://github.com/matchai/spacefish/blob/74dec4a89b946ee9aa36beca8ac1650a9e3efe1f/tests/__sf_section_aws.test.fish#L6

I'd also like to note that on this line just, `cd`, should suffice. It always defaults to home anyways!
https://github.com/matchai/spacefish/blob/74dec4a89b946ee9aa36beca8ac1650a9e3efe1f/tests/__sf_section_dir.test.fish#L31

File: tests/__sf_section_golang.test.fish [hunk 1]
```diff
@@ -3,7 +3,7 @@ source $DIRNAME/mock.fish
 
 function setup
 	spacefish_test_setup
-	mock go 0 "echo \"go version go1.10.3 darwin/amd64\""
+	mock go version 0 "echo \"go version go1.10.3 darwin/amd64\""
```
Comment (Snuggle) [inline tests/__sf_section_golang.test.fish:6]: The same here, did you get variable passing working?
How does it work? :blush:

Comment (matchai) [reply tests/__sf_section_golang.test.fish:6]: Yep! It's documented here:
https://github.com/matchai/spacefish/blob/39e98ec9600714b1f971f74098300a1b95e7f46b/tests/mock.fish#L1-L17

Comment (matchai) [reply tests/__sf_section_golang.test.fish:6]: I'll update it to add usage examples.

Comment (Snuggle) [reply tests/__sf_section_golang.test.fish:6]: Oh!
> ('*' defines a fallback for all arguments) 

Is this the purpose of... `\*`? Should it also be documented that it needs escaping?

Comment (matchai) [reply tests/__sf_section_golang.test.fish:6]: Yeah, I could specify that here. Before I do, do you have any issues with the API as it is? Would something be better suited than `\*`?

**Comment by matchai** [comment]
Be sure to use the handy "No Whitespace" button when reviewing this. The bulk of the diff is really just whitespace changes 😅 

image

File: tests/__sf_section_golang.test.fish (hunk 1 above)
Comment (Snuggle) [reply tests/__sf_section_golang.test.fish:6]: Not that I can think of! Just needs a tiny bit of explaination.
I've usually seen something like `**args` or `$argv` or something. At first glance a wildcard makes me think of file globbing. I don't mind that much though, heh.

In the documentation, I'm not exactly a fan of using `$` for the example command. It implies a prompt or interactive use, seems a little odd. I'd also like to ask for a single blank line between usage information and the code. :yum:

**Comment by Snuggle** [comment]
I cannot see this button? Are you enrolled in one of the fancy GitHub beta experiments?

screenshot_20181209-191230

File: tests/__sf_section_golang.test.fish (hunk 1 above)
Comment (Snuggle) [reply tests/__sf_section_golang.test.fish:6]: @matchai

**Comment by matchai** [comment]
Oh haha. I've been using [Refined GitHub](https://github.com/sindresorhus/refined-github) for so long that I didn't realize that this wasn't a core GitHub feature. 😅
I highly recommend it though!

File: tests/__sf_section_golang.test.fish (hunk 1 above)
Comment (matchai) [reply tests/__sf_section_golang.test.fish:6]: Yep! Totally fair. I'll get rid of the `$`s.

**Review by Snuggle — APPROVED** [review]

**Pull request merged**